### DISCUSSION TITLE
Ingredient subgroups

### DISCUSSION THREAD

**Discussion by stjeanp** [mealie-recipes/mealie#844] [Feature Request]
**Before submitting this feature request I have**
_COMPLETE AND FILL THESE OUT_
- [ x] Reviewed the documentation and verified that my feature request isn't already a feature.
- [x ] Reviewed the existing feature requests and my feature has not been requested.
- [x ] Checked the [tasks taged](https://github.com/hay-kot/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

**Please Describe The Problem To Be Solved**
Some recipes, such as desserts, have a main ingredient list for the cake (for example) and then another list for the glaze or topping. It would be nice to be able to create subgroups of ingredients within one recipe to break that up logically.

Example:

* Cake
* * 1 kg flour
* * 12 eggs
* Frosting
*  * 1 kg sugar
* * 500g butter

**(Optional): Suggest A Solution**

Preferred solution would be to update the recipe schema to allow named groups of ingredients within a recipe to take in to account different groups of ingredients for different parts of a completed food.

The default can be a single group just named "Ingredients". This might require a change to the rendering subsystem so that we don't get a repeated name.

**Additional Information**
_FILL THESE OUT_
- [ ] If the feature is accepted I'm willing to submit a PR to provide this feature
I can't do it right now. With work and family commitments I just don't have the time to sling more code. I'm sorry... :(

**Comment by stjeanp** [comment]
I found this functionality already in the application. The documentation doesn't mention it, so I'd like to keep the request open for a little bit to see if I can update that and send you a PR.

**Reply by Northguy** [reply]
Where did you find this functionality in the application? I do not seem to be able to find it?

**Reply by Northguy** [reply]
Never mind... found it by clicking on the + sign of an existing ingredient

**Reply by epheterson** [reply]
Oh wow, that is hidden, good to know! Might be more obvious if there were a "New Section" button at the bottom or something to turn that feature on and add a second section in one click.
